.pre-commit-config.yaml

```diff
@@ -64,7 +64,7 @@ repos:
           - tomli
 
   - repo: https://github.com/astral-sh/ruff-pre-commit
-    rev: v0.11.12
+    rev: v0.12.0
```

Comment:
while we're at it, can we also update `- id: ruff`, which is a deprecated alias, to `- id: ruff-check` ?

Reply:
Done.